### DISCUSSION TITLE
Task - Spring Boot - Spring JDBC - my_org DB

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/spring-framework#11] [Tasks]
# CRUD API using Spring JDBC, DataSource, and JdbcTemplate

Create a Spring Boot application named `springBootJdbcTemplate` (artifactId and groupId are in all lowercase) to connect with the `org` DB using the **Spring JDBC** module. Create the base package of the project named `com.springboot.workers.springjdbcexample`.

The details for creating the Spring Boot project are as follows:
- name: `springBootJdbcTemplate`
- group: `com.workers`
- artifact: `springbootjdbctemplate`
- package: `com.springboot.workers.springjdbcexample`

Dependencies to include:
- Spring Starter Web
- Spring Starter JDBC
- Spring MySQL
- Spring Boot DevTools (optional)
- Spring Configuration Processor (optional)

    ```xml
    <dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-jdbc</artifactId>
		</dependency>

		<dependency>
			<groupId>mysql</groupId>
			<artifactId>mysql-connector-java</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
    </dependencies>
    ```

Perform _CRUD_ operations on the `Worker` table using models, DAOs, and repositories. 

Create these sub-packages under the base package:
  - `com.springboot.workers.springjdbcexample.model`
  - `com.springboot.workers.springjdbcexample.controller`
  - `com.springboot.workers.springjdbcexample.dao`
  - `com.springboot.workers.springjdbcexample.mapper` (Optional)
  - `com.springboot.workers.springjdbcexample.repository`
  - `com.springboot.workers.springjdbcexample.util`

Create the following **model** classes in `com.springboot.workers.springjdbcexample.controller` package:
- `Worker`

Create the following **dao** interfaces in `com.springboot.workers.springjdbcexample.dao` package:
- `WorkerDAO`

[Optional] Create the following **mapper** classes in `com.springboot.workers.springjdbcexample.mapper` package:
- `WorkerMapper` 

Create the following **repository** classes in `com.springboot.workers.springjdbcexample.repository` package:
- `WorkerRepository`

Create the helper classes such as `DateTimeUtilities` in the `com.springboot.workers.springjdbcexample.util` package`.

**Hint** - You can reuse code snippets from discussion #6 (models, daos, repositories, mappers) and discussion #10 (controllers, utilities, driver)

In Spring Boot, DataSource bean gets initialized automatically if not mentioned explicitly.
If the user does not want this then it can be done by setting the property `spring.datasource.initialize` to `false`. You can configure the DataSource in `application.properties` files under `resources` folder with the following lines:
```
spring.datasource.url=jdbc:mysql://localhost/cameldb
spring.datasource.username=root
spring.datasource.password=rida
spring.datasource.platform=mysql
```

Create the following **controller** classes in `com.springboot.workers.springjdbcexample.controller` package:
- `HomeController`
- `WorkerController`

Create the handler methods and the following mappings in the appropriate controllers:
- `/` - Displays a welcome message "Hello visitor!\nVisiting time: ${CURRENT_TIMESTAMP}"
- `/worker/showWorker` - Displays the details of a `Worker` record. Use a hard-coded value of `worker_id` to call the respective repository function.
- `/worker/all/showWorkers` - Displays the details of all the workers in the `Worker` table
- `/worker/create` - Create a new worker record. Use hard-coded values to call the respective repository function for the same.
- `/worker/update` -Update any worker record's `email` to "mfs.akash@gmail.com". Use hard-coded values to call the respective repository function for the same.
- `/worker/delete` -Delete any worker record. Use hard-coded values to call the respective repository function for the same.

**Comment by abhiraj2k** [comment]
https://github.com/abhiraj2k/spring-projects/tree/master/springbootdemo

**Comment by shreiyarandive** [comment]
https://github.com/shreiyarandive/spring-java/tree/main/springBootJdbcTemplate

**Comment by RohitNaikade1** [comment]
https://github.com/SpringBootProjectsWiley/SpringBootJdbcTemplate.git

**Comment by preeti-max** [comment]
https://github.com/preeti-max/springDemo/tree/master/springBootJdbcTemplate

**Comment by Ashwesh09** [comment]
[My submission](https://github.com/Ashwesh09/Web-Projects/tree/main/springbootjdbctemplate)

**Comment by Sarfinaa** [comment]
[Spring JDBC Template Code](https://github.com/Sarfinaa/SpringAndSpringBootProjects/tree/master/springBootJdbcTemplate)

**Comment by ratnadeepikavuddagiri** [comment]
https://github.com/ratnadeepikavuddagiri/SpringProjects/tree/main/springbootjdbctemplate

**Comment by PavanArvapally** [comment]
https://github.com/PavanArvapally/spring-projects/tree/main/springBootJdbcTemplate

**Comment by RohitN09** [comment]
https://github.com/RohitN09/Spring/tree/master/springBootJdbcTemplate

**Comment by Mrafe** [comment]
<a href="https://github.com/Mrafe/Spring/tree/main/jdbcExample">Spring Boot JDBC Task</a>

**Comment by rakeshGit419** [comment]
[https://github.com/rakeshGit419/Wiley_Java/tree/main/OneDrive/Desktop/Wiley_java/spring%20projects/springbootjdbctemplate](url)

**Comment by Batchusatyasri** [comment]
https://github.com/Batchusatyasri/spring_core/tree/main/springbootjdbctemplate

**Comment by amruth2211** [comment]
https://github.com/amruth2211/Spring-Projects/tree/main/springBootJdbcTemplate/src/main

**Comment by Krishreddy460** [comment]
[jdbcspringboot](https://github.com/Krishreddy460/springbootJDBCTemplate)

**Comment by Harshita-Joshi** [comment]
https://github.com/Harshita-Joshi/Spring/tree/main/springbootjdbctemplate

**Comment by SrideviYamala** [comment]
https://github.com/SrideviYamala/spring-projects/tree/master/springbootjdbctemplate

**Comment by lalit-ky** [comment]
https://github.com/009lalitkumar/Spring-SpringBoot-Tasks/tree/main/11

**Comment by Khyati-tripathi** [comment]
https://github.com/Khyati-tripathi/Spring/tree/main/springBootJdbcTemplate

**Comment by KaminiJha** [comment]
https://github.com/KaminiJha/SpringBootTasks/tree/main/springBootJdbcTemplate

**Comment by Prem-3** [comment]
[https://github.com/Prem-3/Task/tree/main/springbootjdbctemplate](url)

**Comment by ujjwal18bit1167** [comment]
https://github.com/ujjwal18bit1167/Spring-Framework/tree/main/springBootJdbcTemplate
